CMakeLists.txt

```diff
@@ -848,7 +802,6 @@ target_include_directories(
          $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include/>
          $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include/>
          $<INSTALL_INTERFACE:include/>
-         $<BUILD_INTERFACE:$<$<AND:$<BOOL:${COAP_WITH_LIBTINYDTLS}>,$<BOOL:${USE_VENDORED_TINYDTLS}>>:${CMAKE_BINARY_DIR}/include/tinydtls>>
```

Comment:
To be able to build with USE_VENDORED_TINYDTLS set to OFF. you need the following 2 lines here
````
+         $<$<BOOL:${COAP_WITH_LIBTINYDTLS}>:${TINYDTLS_INCLUDE_DIR}>
+         $<$<BOOL:${COAP_WITH_LIBTINYDTLS}>:${TINYDTLS_INCLUDE_DIR}/tinydtls>
````
It appears that the second line is needed as well when building the tinydtls' own subdirectories.